spec/features/bulkrax_dynamic_import_spec.rb

```diff
@@ -0,0 +1,173 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+require "csv"
+
+RSpec.describe "Bulkrax import", clean: true, slow: true do
+  let(:model_name) { "UbiquityTemplateWork" }
+  let(:number_of_records) { 2 }
+
+  let(:user) { create(:user, email: "test@example.com") }
+  let!(:depositor) { build_stubbed(:user, email: "batchuser@example.com") }
+  let(:temporary_file_path) { "spec/fixtures/csv/dynamic_data.csv" }
+
+  let(:importer) do
+    create(:bulkrax_importer_csv,
+           user: user,
+           field_mapping: Bulkrax.field_mappings["HykuAddons::CsvParser"],
+           parser_klass: "HykuAddons::CsvParser",
+           parser_fields: { "import_file_path" => temporary_file_path },
+           limit: 0)
+  end
+
+  before do
+    # Make sure default admin set exists
+    AdminSet.find_or_create_default_admin_set_id
+    stub_request(:get, Addressable::Template.new("#{Hyrax::Hirmeos::MetricsTracker.translation_base_url}/translate?uri=urn:uuid:{id}")).to_return(status: 200)
+    allow(Hyrax::Hirmeos::HirmeosFileUpdaterJob).to receive(:perform_later)
+
+    create_csv
+
+    perform_enqueued_jobs(only: Bulkrax::ImporterJob) do
+      importer.import_collections
+    end
+  end
+
+  after do
+    delete_csv_file
+  end
+
+  it "imports works" do
+    expect do
+      perform_enqueued_jobs(only: Bulkrax::ImporterJob) do
+        importer.import_works
+      end
+    end.to change { model_name.constantize.count }.by(number_of_records)
+
+    aggregate_failures do
+      CSV.read(temporary_file_path, headers: true).each do |row|
+        work = model_name.constantize.where(source_identifier: row["source_identifier"]).first
+        expectations = csv_row_to_expectations(row, work)
+
+        expectations.each do |expectation|
+          expect(expectation[:actual]).to eq(expectation[:test]), "expected #{expectation[:attribute_name]} to equal #{expectation[:test]} but got #{expectation[:actual]}"
+        end
+      end
+    end
+  end
+
+  private
+
+    # Schema methods
+
+    def field_configs
+      Hyrax::UbiquityTemplateWorkForm.field_configs.reject { |k, _v| untested_attributes.include?(k) }
+    end
+
+    def field_configs_without_subfields
+      field_configs.reject { |_k, v| v.key?(:subfields) }
+    end
+
+    def subfields
+      field_configs.map { |_k, v| v.dig(:subfields) }
+                   .compact
+                   .reduce({}, :merge)
+                   .reject { |k, _v| untested_attributes.include?(k) }
+    end
+
+    def attributes
+      field_configs_without_subfields.merge!(subfields)
+    end
+
+    def untested_attributes
+      %i[resource_type language rights_statement relation_type subject display_creator_profile]
```

Comment:
Please comment on whether all of these should be excluded.  The intention here is only to exclude internal fields which cause specific behaviours of the importer or application which should be tested in isolation, eg embargoing, DOI minting. I am unsure for instance why `display_creator_profile` never succeeds.

Reply:
@BertZZ might know more about `display_creator_profile`. I thought it was part of the JSON fields.

Reply:
I will rebase this on main and see if we can remove this once main is passing again.

Reply:
`display_creator_profile` is now no longer required! So we've definitely fixed this one.